include/core/Godot.hpp

```diff
@@ -513,23 +513,28 @@ T *Object::cast_to(const Object *obj) {
 	if (!obj)
 		return nullptr;
 
-	size_t have_tag = (size_t)godot::nativescript_1_1_api->godot_nativescript_get_type_tag(obj->_owner);
-
-	if (have_tag) {
-		if (!godot::_TagDB::is_type_known((size_t)have_tag)) {
-			have_tag = 0;
+	if (T::___CLASS_IS_SCRIPT) {
+		size_t have_tag = (size_t)godot::nativescript_1_1_api->godot_nativescript_get_type_tag(obj->_owner);
+		if (have_tag) {
+			if (!godot::_TagDB::is_type_known((size_t)have_tag)) {
+				have_tag = 0;
+			}
 		}
-	}
 
-	if (!have_tag) {
-		have_tag = obj->_type_tag;
-	}
+		if (!have_tag) {
+			have_tag = obj->_type_tag;
+		}
 
-	if (godot::_TagDB::is_type_compatible(typeid(T).hash_code(), have_tag)) {
-		return (T::___CLASS_IS_SCRIPT) ? detail::get_custom_class_instance<T>(obj) : (T *)obj;
+		if (godot::_TagDB::is_type_compatible(typeid(T).hash_code(), have_tag)) {
```

Comment:
Perhaps we can encapsulate this `typeid(T).hash_code()` by using `__get_id()` instead?